### PR TITLE
Add docusaurus

### PR DESCRIPTION
**Pull request by TheJuanAndOnly99**
#133 This PR adds Docusaurus v3.

### CODE REVIEW

**Comment by psmulovics** [comment]
@mimiflynn, FYI

File: netlify.toml [hunk 1]
```diff
@@ -1,2 +1,2 @@
 [build]
-   ignore = "/bin/false"
\ No newline at end of file
+  ignore = "git diff --quiet $CACHED_COMMIT_REF $COMMIT_REF . ../{docs,website}/"
```
Comment (psmulovics) [inline netlify.toml:2]: @TheJuanAndOnly99 , we tried quite a few combination of this - and ended up just by using /bin/false . If you figure out a working setup (that works for the draft pages too and for normal, etc), I am super interested.

Comment (TheJuanAndOnly99) [reply netlify.toml:2]: Hi @psmulovics not sure I understand what you mean by draft vs normal pages. The objective here is to only run Netlify on PRs that contain changes to the docs and website folders.

**Comment by maoo** [comment]
@DovOps @rocketstack-matt - could you please review this PR, so we can move forward with the merge?

**Comment by rocketstack-matt** [comment]
> @DovOps @rocketstack-matt - could you please review this PR, so we can move forward with the merge?

Is the intent just to get the docs project in the build, as obviously the docs themselves are missing other than the front page? If so the setup looks fine to me, but I can't merge as I don't have write access to the repo. Which raises the question should we be adding CODEOWNERS into this repo and broadening the maintainer list? @maoo / @DovOps

**Comment by maoo** [comment]
> > @DovOps @rocketstack-matt - could you please review this PR, so we can move forward with the merge?
> 
> Is the intent just to get the docs project in the build, as obviously the docs themselves are missing other than the front page? If so the setup looks fine to me, but I can't merge as I don't have write access to the repo. Which raises the question should we be adding CODEOWNERS into this repo and broadening the maintainer list? @maoo / @DovOps

Good points. Here's my proposal:
1. Make Matt a TraderX Maintainer (@DovOps - please lmk if you agree and I'll go ahead with permission changes)
2. Matt raises Pull Request around CODEOWNERS
3. Matt raises an issue to port documentation from github READMEs to Docusaurus (and update links into other collaterals); this issue could be a `Good First Issue`

Thoughts?

**Comment by DovOps** [comment]
@maoo  I suggest we merge this - good with perm changes.  And we can make additional changes once this is properly in the build

**Pull request merged**